backend/apid/routers/cluster.go

```diff
@@ -126,7 +179,18 @@ func (r *ClusterRouter) memberUpdate(w http.ResponseWriter, req *http.Request) {
 }
 
 func (r *ClusterRouter) clusterID(w http.ResponseWriter, req *http.Request) {
-	resp, err := r.controller.ClusterID(req.Context())
+	timeout, err := parseTimeout(req)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	ctx := req.Context()
+	if timeout > 0 {
+		tctx, cancel := context.WithTimeout(ctx, time.Duration(timeout)*time.Second)
+		defer cancel()
+		ctx = tctx
+	}
```

Comment:
This block is repeated so many times: `list()`, `memberUpdate()`, `memberRemove()`, `memberAdd()`, ...
Could it be turned into a function, say `withTimeout(*http.Request) (context, err)`?

Reply:
![image](https://user-images.githubusercontent.com/1633772/70466021-155bf580-1a77-11ea-8b36-dce223c935c1.png)